pkg/store/proxy_test.go

```diff
@@ -573,6 +573,45 @@ func TestProxyStore_LabelValues(t *testing.T) {
 	testutil.Equals(t, 1, len(resp.Warnings))
 }
 
+func TestProxyStore_LabelNames(t *testing.T) {
+	defer leaktest.CheckTimeout(t, 10*time.Second)()
+
+	m1 := &mockedStoreAPI{
+		RespLabelNames: &storepb.LabelNamesResponse{
+			Names:    []string{"a", "b"},
+			Warnings: []string{"warning"},
+		},
+	}
+
+	m2 := &mockedStoreAPI{
+		RespLabelNames: &storepb.LabelNamesResponse{
+			Names: []string{"a", "c", "d"},
+		},
+	}
+
+	cls := []Client{
+		&testClient{StoreClient: m1},
+		&testClient{StoreClient: m2},
+	}
+
+	q := NewProxyStore(nil,
+		func(context.Context) ([]Client, error) { return cls, nil },
+		component.Query,
+		nil,
+	)
+
+	ctx := context.Background()
+	req := &storepb.LabelNamesRequest{
+		PartialResponseDisabled: true,
+	}
+	resp, err := q.LabelNames(ctx, req)
+	testutil.Ok(t, err)
+	testutil.Assert(t, proto.Equal(req, m1.LastLabelNamesReq), "request was not proxied properly to underlying storeAPI: %s vs %s", req, m1.LastLabelNamesReq)
+
+	testutil.Equals(t, []string{"a", "b", "c", "d"}, resp.Names)
```

Comment:
if partial response is disabled this should not return data when warnings occur.